src/cascadia/TerminalApp/AppKeyBindings.cpp

```diff
@@ -1,117 +1,294 @@
+    DEFINE_EVENT(AppKeyBindings, PrevTab,           _PrevTabHandlers,           TerminalApp::PrevTabEventArgs);
+    DEFINE_EVENT(AppKeyBindings, IncreaseFontSize,  _IncreaseFontSizeHandlers,  TerminalApp::IncreaseFontSizeEventArgs);
+    DEFINE_EVENT(AppKeyBindings, DecreaseFontSize,  _DecreaseFontSizeHandlers,  TerminalApp::DecreaseFontSizeEventArgs);
+    DEFINE_EVENT(AppKeyBindings, ScrollUp,          _ScrollUpHandlers,          TerminalApp::ScrollUpEventArgs);
+    DEFINE_EVENT(AppKeyBindings, ScrollDown,        _ScrollDownHandlers,        TerminalApp::ScrollDownEventArgs);
+
+
+    // Method Description:
+    // - Deserialize an AppKeyBindings from the key mappings that are in the
+    //   array `json`. The json array should contain an array of objects with
+    //   both a `command` string and a `keys` array, where `command` i one of
+    //   the names listed in `commandNames`, and `keys` is an array of
+    //   keypresses. Currently, the array should contain a single string, which
+    //   can be deserialized into a KeyChord.
+    // Arguments:
+    // - json: and arrayof JsonObject's to deserialize into our _keyShortcuts mapping.
+    // Return Value:
+    // - the newly constructed AppKeyBindings object.
+    TerminalApp::AppKeyBindings AppKeyBindings::FromJson(Windows::Data::Json::JsonArray const& json)
+    {
+        TerminalApp::AppKeyBindings newBindings{};
+
+        for (const auto& value : json)
+        {
+            if (value.ValueType() == JsonValueType::Object)
+            {
+                JsonObject obj = value.GetObjectW();
+                if (obj.HasKey(COMMAND_KEY) && obj.HasKey(KEYS_KEY))
+                {
+                    const auto commandString = obj.GetNamedString(COMMAND_KEY);
+                    const auto keys = obj.GetNamedArray(KEYS_KEY);
+                    if (keys.Size() != 1)
+                    {
+                        continue;
+                    }
+                    const auto keyChordString = keys.GetAt(0).GetString();
+                    ShortcutAction action;
+                    bool parsedSuccessfully = false;
+                    // Try matching the command to one we have
+                    for (const auto& pair : commandNames)
+                    {
+                        if (pair.second == commandString)
+                        {
+                            action = pair.first;
+                            parsedSuccessfully = true;
+                            break;
+                        }
+                    }
+                    if (!parsedSuccessfully)
+                    {
+                        continue;
+                    }
+                    // Try parsing the chord
+                    Settings::KeyChord chord{ false, false, false, 0 };
+                    parsedSuccessfully = false;
+                    try
+                    {
+                        chord = Settings::KeyChord::FromString(keyChordString);

```

Comment:
these should probably be folded.. we don't need to create a temporary chord, we can bail immediately if it's parsed successfully, and `SetKeyBinding` in the try